[charts] Add bar gradient example (#19174)
mui/mui-x: docs/data/charts/bars/bars.md

{{"demo": "BarGradient.js"}}

### Gradients

By default, a gradient's units are set to `objectBoundingBox`.

> When applied to a bar, the gradient will stretch to fill the entire size of the bar, regardless of the bar's value.

Reviewer: ```suggestion
When applied to a bar, the gradient stretches to fill the entire size of the bar, regardless of the bar's value.
```

> Alternatively, you can set `gradientUnits` to `userSpaceOnUse`, which stretches the gradient to fill the entire size of the chart.

Reviewer: Would probably be nice to have a few words on how to achieve that and how to account for the legend/tooltip

Author: > how to achieve that

Sure, I'll expand on this.

> how to account for the legend/tooltip

No need to account for those since they are outside the SVG. We only need to account for the axes sizes + margins

Author: Done, let me know if it's clearer or if I should rewrite it. I found it a bit hard to explain succintly

> This effectively reveals the gradient depending on the bar's value, as the gradient will be clipped to the bar's size.

Reviewer: ```suggestion
This effectively reveals the gradient depending on the bar's value, as the gradient is clipped to the bar's size.
```
